### PR TITLE
feat: add image and manifest for windows npm

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

build and deploy windows npm example

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: npm/cmd/start.go [hunk 1]
```diff
@@ -58,36 +60,48 @@ func newStartNPMCmd() *cobra.Command {
...
 	return startNPMCmd
 }
 
-func start(config npmconfig.Config) error {
+func start(config npmconfig.Config, flags npmconfig.Flags) error {
 	klog.Infof("loaded config: %+v", config)
 	klog.Infof("Start NPM version: %s", version)
 
 	var err error
-	defer func() {
```
Comment (vakalapa) [inline npm/cmd/start.go:75]: Is there a reason to remove recover ?

File: npm/cmd/start.go [hunk 2]
```diff
@@ -58,36 +60,48 @@ func newStartNPMCmd() *cobra.Command {
...
 	metrics.InitializeAll()
 
-	// Creates the in-cluster config
-	k8sConfig, err := rest.InClusterConfig()
-	if err != nil {
-		return fmt.Errorf("failed to load in cluster config: %w", err)
+	// Create the kubernetes client
+	var k8sConfig *rest.Config
+	if flags.KubeConfigPath == "" {
+		var err error
```
Comment (vakalapa) [inline npm/cmd/start.go:94]: move this var err outside the if condition?

File: npm/examples/windows/setkubeconfigpath.ps1 [hunk 3]
```diff
@@ -0,0 +1,9 @@
+$cpEndpoint = Get-Content C:\k\config | ForEach-Object -Process {if($_.Contains("server:")) {$_.Trim().Split()[1]}}
```
Comment (vakalapa) [inline npm/examples/windows/setkubeconfigpath.ps1:1]: Can you add some comments on what/how cpendpoint is or looks like, similarly for other fields. Have a readme.md for this logic, if possible

File: npm/cmd/start.go [hunk 4]
```diff
@@ -37,11 +38,12 @@ func newStartNPMCmd() *cobra.Command {
 			viper.AutomaticEnv() // read in environment variables that match
 			viper.SetDefault(npmconfig.ConfigEnvPath, npmconfig.GetConfigPath())
 			cfgFile := viper.GetString(npmconfig.ConfigEnvPath)
+
 			viper.SetConfigFile(cfgFile)
 
 			// If a config file is found, read it in.
 			if err := viper.ReadInConfig(); err == nil {
```
Comment (vakalapa) [inline npm/cmd/start.go:45]: Add a comment here saying there is no config merging, and change in bhevaior needs an explicit true for backward compability.

**Comment by matmerr** [comment]
add powershell.exe -command "& { . .\windows.ps1; azure-npm-image v26-windows-amd64 }"

File: npm/config/config.go [hunk 5]
```diff
@@ -34,3 +34,7 @@ type Toggles struct {
 	EnableHTTPDebugAPI      bool
 	EnableV2Controllers     bool
 }
+
+type Flags struct {
+	KubeConfigPath string `json:"KubeConfigPath"`
```
[the remark this answers is not in the document]
Comment (matmerr) [reply npm/config/config.go:39]: it makes the signature for start cleaner, passing in config struct and flags struct, when we add more flags in the future

`func start(config npmconfig.Config, flags npmconfig.Flags) error {`

File: npm/cmd/start.go (hunk 1 above)
Comment (matmerr) [reply npm/cmd/start.go:75]: if we have a crash we want to crash hard and bubble that up to orchestrator

File: npm.txt [hunk 6]
```diff
@@ -0,0 +1,59 @@
+start here
```
Comment (matmerr) [inline npm.txt:1]: remove this

File: npm/cmd/start.go [hunk 7]
```diff
@@ -97,6 +114,8 @@ func start(config npmconfig.Config) error {
 		return fmt.Errorf("failed to generate clientset with cluster config: %w", err)
 	}
 
+	klog.Infof("received clientset %+v", clientset)
```
Comment (matmerr) [inline npm/cmd/start.go:117]: delete this from logs

File: npm/azure-npm.yaml [hunk 8]
```diff
@@ -34,7 +34,7 @@ rules:
       - watch
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
+kind: ClusterRoleBinding
```
Comment (vakalapa) [inline npm/azure-npm.yaml:37]: <Not> remove the space,

File: npm/cmd/main.go [hunk 9]
```diff
@@ -3,13 +3,61 @@
...
 
+var flagDefaults = map[string]string{
+	flagKubeConfigPath: "",
+}
+
 // Version is populated by make during build.
 var version string
 
 func main() {
+	fmt.Println("start here")
```
Comment (vakalapa) [inline npm/cmd/main.go:26]: we will need to remove these comments in next PR

File: npm/cmd/main.go [hunk 10]
```diff
@@ -3,13 +3,61 @@
...
+	if version != "" {
+		viper.Set(flagVersion, version)
+	}
+
+	fmt.Println("made it here")
+	cobra.OnInitialize(func() {
+		viper.AutomaticEnv()
+		initCommandFlags(rootCmd.Commands())
+	})
+	fmt.Println("made it here2")
```
Comment (vakalapa) [inline npm/cmd/main.go:39]: this should be removed too.

**Review by vakalapa — APPROVED** [review]
lgtm just couple of comments for us to apply in next PR

**Review by vakalapa — APPROVED** [review]

**Pull request merged**